app/src/main/java/edu/usf/sas/pal/muser/firebase/models/CurrentSongInfo.kt

```diff
@@ -0,0 +1,32 @@
+package edu.usf.sas.pal.muser.firebase.models
+
+
+/**
+ * A model class that holds the information about the song that is currently being played by the
+ * user.
+ */
+data class CurrentSongInfo
+
+/**
```

Comment:
https://github.com/CUTR-at-USF/MUSER/issues/17#issuecomment-659595447 contains a longer list of fields that are currently in the `Song` class. Why aren't those included here? 

Also - could you please list the additional fields here that can be extracted by the ContentProvider (see [this doc](https://docs.google.com/document/d/1QRtP0eO1rrT0JTR6QNPO7kNX-UpHv5CrWZQXa1ZSETE/edit)) but aren't currently contained in the Shuttle `Song` class so we can decide if we want to implement the code to extract them for our purposes?

Reply:
Fields like Album ID and Artist ID are locally created and I don't know if it'd have anything to do with the user. Please let me know if otherwise. 
Playlist order and Playlist Song ID is again locally created and order depends upon the user. It may change if the user shuffles the playlist. 
I didn't include start and elapsed time because, when the track info is updated, the start and elapsed time are always 0.00. Please feel free to let me know if you'd like to add them.  
I can add the bitrate and file size to the SongData classes.

Reply:
The Content provider extracts :- 

- Count
- Content Type
- Data
- Date Modified
- Default Sort Order
- Display Name
- Is Ringtone
- Is Music
- Is Alarm Sound
- Record Sound Action
- Title Key
- Title

Reply:
>Fields like Album ID and Artist ID are locally created and I don't know if it'd have anything to do with the user. Please let me know if otherwise.

Let's store all locally-generated IDs in Firebase. We may need them if we do analysis server-side and then want to provide the user info back on that. For example, if we use machine learning to generate a custom playlist server-side, we'll need to know the song IDs when we fetch that playlist from the Android app so we know which songs to play locally.

>The Content provider extracts :-

What are the definition of these fields? I don't see them in the [MediaStore list](https://developer.android.com/reference/android/provider/MediaStore).

Seems like these could be useful without knowing the definitions:
* Content Type
* Data Modified
* Default Sort Order
* Display Name
* Title Key
* Title

Reply:
>I didn't include start and elapsed time because, when the track info is updated, the start and elapsed time are always 0.00. Please feel free to let me know if you'd like to add them.

So if some skips to the middle of a song using the slider, the start and elapsed time don't change? Of if they start a song, stop in the middle of it and pick up in the app later to restart in the middle, it doesn't change?

I think we'll want to change that behavior if so. But that would likely go in the "actions" class, not the song class.

Reply:
>I can add the bitrate and file size to the SongData classes.

Yes, let's keep that, thanks.

Reply:
> So if someone skips to the middle of a song using the slider, the start and elapsed time don't change? Of if they start a song, stop in the middle of it and pick up in the app later to restart in the middle, it doesn't change?

Yes, it does. However, I think it makes sense to add it with the `SongAction` class rather than adding it to `SongData`. Initially, I didn't keep the `SongAction` class in mind while making the list and that's probably why I excluded them. 

> What are the definition of these fields? I don't see them in the MediaStore list.
Seems like these could be useful without knowing the definitions:

Please check out these. https://developer.android.com/reference/android/provider/MediaStore.Audio.Media, 
https://developer.android.com/reference/android/provider/MediaStore.Audio.AudioColumns. I found them very helpful in finding definitions for the fields. 

> Let's store all locally-generated IDs in Firebase. We may need them if we do analysis server-side and then want to provide the user info back on that. For example, if we use machine learning to generate a custom playlist server-side, we'll need to know the song IDs when we fetch that playlist from the Android app so we know which songs to play locally.

Sounds great. Will add them to the class.

Reply:
> I think it makes sense to add it with the SongAction class rather than adding it to SongData.

Sounds good!

Reply:
>I found them very helpful in finding definitions for the fields.

Thanks! I'm still not seeing definitions for these fields, though:
* Count
* Data

Reply:
Oops - You can find the definition of `MediaStore.Audio.Media.DATA` here:- https://developer.android.com/reference/android/provider/MediaStore.MediaColumns 

I think `COUNT` returns the number of rows extracted by the content provider when scanning from the storage. I learned it somewhere from the Android documentation a few weeks back but I'm not able to find a link to it. I'll let you know as soon as I find it.

Reply:
Looks like we shouldn't include `DATA`, as it's deprecated - https://developer.android.com/reference/android/provider/MediaStore.MediaColumns#DATA.

Ok, sound like we don't need `COUNT` either.

Reply:
Shuttle currently uses `MediaStore. Audio.Media. DATA` to access the file using the direct file path returned by the same. I found a couple of alternatives to this. 
https://developer.android.com/reference/android/content/ContentResolver#openFileDescriptor(android.net.Uri,%20java.lang.String) (Suggested by Android)
https://stackoverflow.com/questions/57685725/mediastore-mediacolumns-data-is-deprecated-and-i-wanna-load-images-from-gallery (Kotlin based solution looks more straightforward)

Applying the latter in our code, it looks more like:-

Current:-
`path = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Audio.Media.DATA));`

New:-
`path = Uri.withAppendedPath(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI, "" + id)`  (Here `id` is the SongID)

Also, I verified that the function is available in JAVA as well.

Reply:
@BumbleFlash Thanks for looking at this. Let's actually put off changing this for now - could you please open a new issue saying that `DATA` is deprecated and we will eventually need to change it, and link to the above comment?

So for now I guess we should capture the `DATA` element in Firebase, and we can switch to the new method when that's implemented at a later date.

Reply:
Sounds great!